### DISCUSSION TITLE
Resolved: No module named 'ussl' in MicroPython v1.23 or 1.24 preview

### DISCUSSION THREAD

**Discussion by mikespolk** [micropython/micropython#15229] [Libraries & Drivers] [answered]
I have been using the ussl module in RPi Pico script, with MicroPython v1.22. However, after starting to use 1.23 version, I get error message **no module named 'ussl'**.

Non-working versions:

- MicroPython v1.24.0-preview.32.g80a4f632e on 2024-06-08; Raspberry Pi Pico W with RP2040
- MicroPython v1.23.0 on 2024-06-02; Raspberry Pi Pico W with RP2040

Works OK here:
- MicroPython v1.22.1 on 2024-01-05; Raspberry Pi Pico W with RP2040

How to get ussl working again?

**Comment by ekspla** [comment]
Please see this [announcement](https://github.com/orgs/micropython/discussions/15172).

**Reply by robert-hh** [reply, marked as answer]
Try `import ssl`.

**Comment by mikespolk** [comment]
Oh, that's it. Simply replacing the **import ussl** commands with **import ssl** helped and now the code is running successfully in 1.23. Thank you for quick answers!

**Comment by sgbaird** [comment]
Noting in v1.27.0 that there may be an issue with importing both `ssl` and `ussl`: https://github.com/vertical-cloud-lab/byu-vcl/issues/33#issuecomment-4121890075

**Reply by projectgus** [reply]
EDIT: Moved the part that's not relevant here to its own answer.

There are some issues with `import ussl` on recent MicroPython (v1.23 and newer it doesn't work), but in the comment you linked `import ssl` is failing. This might be because that person is accidentally using a build for a board without any `ssl` module, i.e. `RPI_PICO2` instead of `RPI_PICO2_W`.

Since v1.22 MicroPython has deprecated the u-name modules, so unless you need support for very old MicroPython then you should be able to replace `ussl` with `ssl` everywhere.

**Comment by projectgus** [comment]
FWIW, [MicroPython v1.21.0](https://github.com/micropython/micropython/releases/tag/v1.21.0) was the release that renamed all u-prefix modules to their CPython name and deprecated the u-prefix. It's safe to replace `ussl` with `ssl` everywhere, for example.

Backwards compatibility was added for the old names, for example I can do something like `import uarray` and get `array`.

It looks like `import ussl` was accidentally broken in v1.23.0 following this change: https://github.com/micropython/micropython/pull/12259/changes/b802f0f8abf3c198d4d373d934f4ab13ec1552f9

I don't know if that's something we should restore or not, or whether we actually should consider removing all of the backwards compatible code for u-names now.

**Reply by projectgus** [reply]
> whether we actually should consider removing all of the backwards compatible code for u-names now.

@dpgeorge has reminded me that we use the `import u-name` pattern for internal reasons inside micropython-lib in some places, where we want to extend a built-in module with extra Python code via a `mip install` - the Python module named (for example) `socket.py` will `import usocket` and then extend it. This means any other code can `import socket`.

(We don't need this pattern for `import ssl`, because of the change linked above.)

**Reply by projectgus** [reply]
TL;DR - it's recommended to never use `import uname` in MicroPython code, use `import name` instead.
